### DISCUSSION TITLE
How to get the Download plot as png button without `raw_plotly_config: true`??

### DISCUSSION THREAD

**Discussion by deltamelter** [dbuezas/lovelace-plotly-graph-card#486] [Q&A] [answered]
I noticed when checking out a yaml for another chart that it had this option to take a screenshot of the chart.  This option appears because the yaml contains the line:
`raw_plotly_config: true`
This is fine on simple charts, but I tend to overcomplicate mine 🤣 and adding this line to a chart which already has a layout just causes it to mess up.
Would love to just take a snapshot for sharing.  I know this can probably be done with custom button function, but would be cleaner to just enable this built in...  Any ideas how to do that?

![image](https://github.com/user-attachments/assets/b640e222-229f-451c-bfa3-ea6dd7e36359)

**Comment by dbuezas** [comment, marked as answer]
You can override `config` to change the buttons without using `raw_plotly_config`.
The default is:
```yaml
config:
  modeBarButtonsToRemove: ["resetScale2d", "toImage", "lasso2d", "select2d"]
```

([all defaults here](https://github.com/dbuezas/lovelace-plotly-graph-card/blob/master/src/parse-config/defaults.ts))

So you could do this:
```yaml
config:
  modeBarButtonsToRemove: ["resetScale2d", "lasso2d", "select2d"]
```
which will not remove the `toImage` button

**Comment by deltamelter** [comment]
Perfect, Thanks!!
